pkg/customMetrics/customMetrics.go

```diff
@@ -0,0 +1,105 @@
+package customMetrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Registry-namespaces of operator sources.
+const (
+	REDHAT_OPERATOR_NS    = "redhat-operators"
+	COMMUNITY_OPERATOR_NS = "community-operators"
+	CERTIFIED_OPERATOR_NS = "certified-operators"
+	REGISTRY_NAMESPACE    = "registry_ns"
+	OPSRC_NAME            = "opsrc_name"
+)
+
+// Metric variables that report the error based on the operator-source
+// which fails to reconcile.
+var (
+	redhatErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "redhat_operator_reconcile_failure_count",
+			Help: "Monotonic count of times redhat-opsrc was unable to reconcile",
+		},
+		[]string{REGISTRY_NAMESPACE, OPSRC_NAME},
+	)
+	communityErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "community_operator_reconcile_failure_count",
+			Help: "Monotonic count of times community-opsrc was unable to reconcile",
+		},
+		[]string{REGISTRY_NAMESPACE, OPSRC_NAME},
+	)
+	certifiedErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "certified_operator_reconcile_failure_count",
+			Help: "Monotonic count of times certified-opsrc was unable to reconcile",
+		},
+		[]string{REGISTRY_NAMESPACE, OPSRC_NAME},
+	)
+	customOperatorErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_operator_reconcile_failure_count",
+			Help: "Monotonic count of any of the custom operator was unable to reconcile",
+		},
+		[]string{OPSRC_NAME},
+	)
+
+	// List of custom metrics which are to be collected.
+	metricsList = []prometheus.Collector{
+		redhatErrorCount,
+		communityErrorCount,
+		certifiedErrorCount,
+		customOperatorErrorCount,
+	}
+
+	// custom metrics port and path where the metrics would be
+	// exposed.
+	metricsPath = "/metrics"
+	metricsPort = ":8383"
```

Comment:
Same as above... Why is this described in two places? You could probably make the `metricsPath, metricsPort, and metricsHost` global constants within the metrics package.